### PR TITLE
Rewrite ALSA buffering

### PR DESCRIPTION
**Pull request by aykevl**
This PR fixes two main issues:

  * On some hardware, the output stutters a bit. This was the case on my laptop (Fedora Asahi) which despite being very fast still couldn't play audio without crackling on the laptop speakers (it did work over USB though). This was fixed by setting the period size to bufferSize/4, which was previously set to a default (very short) duration.  
    I suspect this is the root cause of https://community.volumio.com/t/new-2023-spotify-plugin/63381/354
  * When pausing and resuming (possibly with a seek in between), some audio from before the pause would be repeated. This wasn't very noticeable with the previous short period size (80ms or 160ms or so), but with the new period size it suddenly became 2 seconds and _very_ noticeable.  
    An easy way to test this bug is to play a loud part of a song, pause, seek to a quiet part (the beginning is usually quiet), and resume. Before this PR, there was a bit of a stutter. After this PR, the stutter is gone entirely.

Note that this PR doesn't close/reopen the output device anymore on every pause/resume. I don't know whether anybody relied on this behavior but it's a change so noting it just in case.

### CODE REVIEW

File: output/driver_unix.go [hunk 1]
```diff
@@ -244,102 +231,96 @@ func (out *output) logParams(params *C.snd_pcm_hw_params_t) error {
...
+			return err
 		}
 
 		out.cond.L.Unlock()
 	}
 }
 
 func (out *output) Pause() error {
-	// Do not use snd_pcm_drop as this might hang (https://github.com/libsdl-org/SDL/blob/a5c610b0a3857d3138f3f3da1f6dc3172c5ea4a8/src/audio/alsa/SDL_alsa_audio.c#L478).
-
```
[the remark this answers is not in the document]
Comment (devgianlu) [reply output/driver_unix.go:319]: This comes from [ebiten](https://github.com/ebitengine/oto/blob/03f5f70bd121e0cbd0faedcf994f54955fa90e42/driver_unix.go#L254), the code I used as inspiration to write this one.

**Comment by devgianlu** [comment]
> Note that this PR doesn't close/reopen the output device anymore on every pause/resume. I don't know whether anybody relied on this behavior but it's a change so noting it just in case.

This is required when working on systems that have multiple players. Imagine that I have go-librespot for Spotify and mpd for local files: if I want to switch from one player to the other it's easier if they release the device when pausing so that the other one can takeover without them having to coordinate.

**Comment by aykevl** [comment]
Right, I see. I will update the PR to restore the previous behavior of closing and reopening the ALSA device on every pause/resume.

(My personal opinion is that this should not be handled by applications, but rather by an audio server like PulseAudio or PipeWire. They are designed to fill this exact role. But I checked, and MPD indeed closes the audio device on pause when using the ALSA output).

**Comment by aykevl** [comment]
Updated to close and restore PCM on pause/resume.

The behavior is still slightly different from what it was before in a different way: it is now only opened when it is first resumed (so it isn't opened directly after connecting to go-librespot). This is consistent with mpd for example and makes more sense to me, because it won't lead to unexpected "ALSA device is busy" errors after connecting but not playing any music.

File: output/driver_unix.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aykevl) [reply output/driver_unix.go:319]: I see. I think we can ignore it (and it sounds more like an ALSA/driver bug anyway).

**Pull request merged**